### ISSUE TITLE
fix(data-sharing): automatic organisation sharing bug when the user is part of the platform main organization

### ISSUE THREAD

**Issue by Lhorus6** [LuatixHQ/opencti#4836]
## Description

If a user is part of the platform main organization, the entities they create will not be shared with the organizations to which they belong. Let me explain with this scenario:

### Doesn't work

Configuration :
- **Org A is the platform main organization**
- User A part of Org A, B and C
- User B part of Org B

Scenario : 
User A create a report (same with malware for example) named "Test report" -> **User B can't see** the report + if User A look next to the orange share button on the report, he doesn't see "org B" and "Org C" as organizations with which the report is shared

### Works

Configuration :
- **Neither Org A, B nor C are the platform main organization**
- User A part of Org A, B and C
- User B part of Org B

Scenario : 
User A create a report (same with malware for example) named "Test report" -> **User B can see** the report + if User A look next to the orange share button on the report, he see "Org A", "Org B" and "Org C" as organizations with which the report is shared (as expected)

## Environment

OpenCTI 5.11.12

## Reproducible Steps

See the description

**Comment by Kedae** [comment]
@Jipegien or @SamuelHassine Need your opinion

**Comment by richard-julien** [comment]
My feeling.
Sharing information to platform organizations is useless.
Auto sharing with all the organizations of the users (except the platform one) seems ok for me.
So:
- If user is part of A,B,C (with A as platforme orga), the element should be shared with B and C.
- If user is part of A (with A as platforme orga), the element should not be shared
- If user is part of B, the element should be shared with B

**Comment by Jipegien** [comment]
I am ok with that!

**Comment by SamuelHassine** [comment]
@Jipegien @nino-filigran is it something scheduled for 5.13?

**Comment by nino-filigran** [comment]
No, it is not @SamuelHassine. Is that something to consider according to you?

**Comment by SamuelHassine** [comment]
Yes, this is a tiny adjustment which is not requiring a lot of technical work but which aligns the behavior of the platform with a more meaningful usage.

**Comment by nino-filigran** [comment]
I'll discuss with @Jipegien on how/if we can fits this in and keep you posted on this ticket.

**Comment by nino-filigran** [comment]
@SamuelHassine the content of the 5.13 is quite packed at the moment. We would rather add it on the top of the list for the 5.14.

**Comment by Jipegien** [comment]
@nino-filigran I don't see this issue in the feature backlog. Did you add it?

**Comment by nino-filigran** [comment]
Great catch, it was on my todo. I'll do it this afternoon
